include/dqlite.h

```diff
@@ -17,8 +17,307 @@
+ * longer special and this function is a no-op.
+ */
+int dqlite_server_set_bootstrap(dqlite_server *server);
+
+/**
+ * Declare the address of an existing server in the cluster, which should
+ * already be running.
+ *
+ * This function
+ * multiple times with different addresses. The server addresses declared with
+ * this function will not be used unless @server is starting up for the first
+ * time; after the first startup, the list of servers stored on disk will be
+ * used instead. (It is harmless to call this function unconditionally.)
+ */
+int dqlite_server_set_peer_address(dqlite_server *server, const char *addr);
+
+/**
+ * Configure @server to listen on the address @addr for incoming connections
+ * (from clients and other servers).
+ *
+ * If no bind address is configured with this function, the abstract address
+ * passed to dqlite_server_create will be used. The point of this function is to
+ * support decoupling the abstract address from the networking implementation
+ * (for example, if a proxy is going to be used).
+ *
+ * @addr must use one of the following formats:
+ *
+ * 1. "<HOST>"
+ * 2. "<HOST>:<PORT>"
+ * 3. "@<PATH>"
+ *
+ * Where <HOST> is a numeric IPv4/IPv6 address, <PORT> is a port number, and
+ * <PATH> is an abstract Unix socket path. The port number defaults to 8080 if
+ * not specified. In the second form, if <HOST> is an IPv6 address, it must be
+ * enclosed in square brackets "[]". In the third form, if <PATH> is empty, the
+ * implementation will automatically select an available abstract Unix socket
+ * path.
+ *
+ * If an abstract Unix socket is used, the server will accept only
+ * connections originating from the same process.
+ */
+int dqlite_server_set_bind_address(dqlite_server *server, const char *addr);
+
+/**
+ * Configure the function that this server will use to connect to other servers.
+ *
+ * The same function will be used by the server's attached client to establish
+ * connections to all servers in the cluster. @arg is a user data parameter that
+ * will be passed to all invocations of the connect function.
+ */
+int dqlite_server_set_connect_func(dqlite_server *server,
+				   dqlite_connect_func f,
+				   void *arg);
+
+/**
+ * Configure the target number of voting nodes for this server's cluster.
+ *
+ * When this server is the cluster leader, it will try to promote and demote
+ * other servers to reach the target voter count. You should configure the same
+ * target on all servers that will participate in the cluster.
+ */
+int dqlite_server_set_target_voters(dqlite_server *server, int n);
+
+/**
+ * Configure the target number of standby nodes for this server's cluster.
+ *
+ * When this server is the cluster leader, it will try to promote and demote
+ * other servers to reach the target standby count. You should configure the
+ * same target on all servers that will participate in the cluster.
+ */
+int dqlite_server_set_target_standbys(dqlite_server *server, int n);
```

Comment:
It would probably make sense to change `int n` into `unsigned n`, since this value is supposed to be positive (same for `dqlite_server_set_target_voters`).